src/ucp/wireup/wireup.c

```diff
@@ -1351,28 +1350,70 @@ static void ucp_wireup_discard_uct_eps(ucp_ep_h ep, uct_ep_h *uct_eps,
     }
 }
 
+static int
+ucp_wireup_is_am_lane_replaced(ucp_ep_h ep,
+                               const ucp_lane_index_t *reuse_lane_map)
+{
+    return !ucp_ep_has_cm_lane(ep) && (ep->am_lane != UCP_NULL_LANE) &&
+           (reuse_lane_map[ep->am_lane] == UCP_NULL_LANE) &&
+           !ucp_wireup_ep_test(ucp_ep_get_lane(ep, ep->am_lane));
+}
+
 static int
 ucp_wireup_check_is_reconfigurable(ucp_ep_h ep,
                                    const ucp_ep_config_key_t *new_key,
                                    const ucp_unpacked_address_t *remote_address,
                                    const unsigned *addr_indices)
 {
-    ucp_lane_index_t lane;
+    ucp_lane_index_t lane, wireup_lane, reuse_lane_map[UCP_MAX_LANES];
+    const ucp_ep_config_key_t *old_key;
 
     if ((ep->cfg_index == UCP_WORKER_CFG_INDEX_NULL) ||
         ucp_ep_has_cm_lane(ep)) {
         return 1;
     }
 
-    /* TODO: Support reconfiguration when lanes are created without a wireup_ep
-     * wrapper */
-    for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
-        if (!ucp_wireup_ep_test(ucp_ep_get_lane(ep, lane))) {
+    old_key = &ucp_ep_config(ep)->key;
+
+    /* TODO: 1) Support lanes which are connected to the same remote MD, but
+     *          different remote sys_dev (eg. TCP). */
+    for (lane = 0; lane < old_key->num_lanes; ++lane) {
+        if ((ucp_ep_config_find_match_lane(old_key, lane, new_key) !=
+             UCP_NULL_LANE) &&
+            !ucp_ep_config_lane_is_equal(old_key, new_key, lane)) {
             return 0;
         }
     }
 
-    return 1;
+    ucp_ep_config_lanes_intersect(old_key, new_key, ep, remote_address,
+                                  addr_indices, reuse_lane_map);
+    wireup_lane = ucp_wireup_get_msg_lane(ep, UCP_WIREUP_MSG_REQUEST);
+
+    /* TODO: 2) Support reconfiguration for separated wireup and AM lanes
+     *          during wireup process (request sent). */
+    return !(ep->flags & UCP_EP_FLAG_CONNECT_REQ_QUEUED) ||
+           (ep->am_lane == wireup_lane) ||
+           !ucp_wireup_is_am_lane_replaced(ep, reuse_lane_map);
```

Comment:
so, from my understanding this PR firstly has the deal with wireup_msg_lane, not an am_lane, then need to rename all related things, like `is_am_replaced` -> `is_wireup_msg_replaced`, use corresponding lane access wrapper, etc

Reply:
actually it's also deals with different AM lane and wireup lane, provided that wireup process was completed.

Reply:
In the code I see changes related to `ep->am_lane` which is equals to `wireup_msg_lane` in most particular configurations. And this case is implemented in this PR but common case is in TODO (2)
pls add a summary to the PR, probably my understanding of the change is wrong